lib/internal/Magento/Framework/Data/Test/Unit/CollectionTest.php

```diff
@@ -100,6 +100,15 @@ public function testFlag()
     public function testGetCurPage()
     {
         $this->_model->setCurPage(10);
+        $this->assertEquals(10, $this->_model->getCurPage());
+
+        $this->_model->setCurPage(-10);
+        $this->assertEquals(1, $this->_model->getCurPage());
+
+        $this->_model->setCurPage(0);
+        $this->assertEquals(1, $this->_model->getCurPage());
+
+        $this->_model->setCurPage(1);
```

Comment:
This change removes the assumption that Page 10  of 1 should return Page 1, but preserves the assumption that Pages < 1 return Page 1.